### PR TITLE
V144

### PR DESCRIPTION
**Pull request by jgliss**

### CODE REVIEW

File: pyplis/imagelists.py [hunk 1]
```diff
@@ -1608,21 +1608,15 @@ def _iter_num(self, start_idx, stop_idx):
...
 
     def _last_file(self):
         """Get last file path of image list."""
-        try:
-            return self.files[self.nof - 1]
-        except IndexError:
-            logger.warning("Filelist empty...")
-        except:
-            raise
+        if not bool(self.files):
```
Comment (johannjacobsohn) [inline pyplis/imagelists.py]: are you sure you actually need that book? Wouldn't that be cast implicitly?

**Comment by johannjacobsohn** [comment]
Is there a specific reason to check for and work around missing PIL? Why not just make it a dependency? AFAIK there aren't much unit or high level tests to ensure pyplis is working with/without PIL, so this leads to a lot of untested code paths.

File: pyplis/imagelists.py (hunk 1 above)
Comment (jgliss) [reply pyplis/imagelists.py]: It would :) But this way it will throw an easier to understand exception (at least that's what I hope) ;)

**Comment by jgliss** [comment]
@johannjacobsohn PIL is only required for reading of Heidelberg camera data, and I prefer this not to be an additional dependency for pyplis. I moved the import into the corresponding reading method, and I think that should be fine.

**Pull request merged**